Sofa/framework/Core/src/sofa/core/visual/VisualModel.cpp

```diff
@@ -21,10 +21,34 @@
 ******************************************************************************/
 #include <sofa/core/visual/VisualModel.h>
 #include <sofa/core/objectmodel/BaseNode.h>
+#include <sofa/core/visual/VisualParams.h>
 
 namespace sofa::core::visual
 {
 
+VisualModel::VisualModel():
+    d_draw(initData(&d_draw, true,  "draw", "Display the object or not"))
+{}
+
+void VisualModel::drawVisual(const VisualParams* vparams)
+{
+    // don't draw if specified not to do so in the user interface
+    if (!vparams->displayFlags().getShowVisualModels())
+        return;
+
+    // don't draw if the component is not in valid state
+    if( d_componentState.getValue() == sofa::core::objectmodel::ComponentState::Invalid )
+        return;
+
+    // don't draw if this component is specifically configured to be disabled
+    if (!d_draw.getValue())
+        return;
+
+    const auto stateLifeCycle = vparams->drawTool()->makeStateLifeCycle();
+
+    doDrawVisual(vparams);
```

Comment:
```suggestion
    if (vparams->displayFlags().getShowWireFrame())
    {
        vparams->drawTool()->setPolygonMode(0, true);
    }

    doDrawVisual(vparams);
```
What do you think about enforcing wireframe rendering at the VisualModel level?

Reply:
It is a good idea that naturally fit into the generalized deployement of NVI. Do you want me to add it in this PR ?